### ISSUE TITLE
Question on implementation

### ISSUE THREAD

**Issue by tuhinsubhrahazra** [tradingview/charting-library-examples#365]
I am currently working on a TradingView library related project and here are a few questions related to the TradingView library. 

1. Does the total tick data length have to be more than or equals to the - periodParams.countBack value in getBars function ?
If yes then can i update the value of periodParams.countBack according to the tick data length ?

2. When I scroll horizontally to the left and when I reach the last bar, the 'getBars' function is called infinitely. Why is this happening?

3. Does any Android library exist that for advanced trading view library ?
because when i open the trading view of groww app , my be i wrong but i think it is not implemented in web view.

**Comment by romfrancois** [comment]
>Does the total tick data length have to be more than or equals to the - periodParams.countBack value in getBars function ?

Simplest answer would be no, but I would encourage you to firsyt read the dedicated [documentation](https://www.tradingview.com/charting-library-docs/latest/connecting_data/Datafeed-API) explaining how the datafeed works.

>When I scroll horizontally to the left and when I reach the last bar, the 'getBars' function is called infinitely. Why is this happening?

This is simply because your datafeed is not indicating to the library there is no more data to return so as a consequence the library keeps on requesting data.

Have a read at this documentation [article](https://www.tradingview.com/charting-library-docs/latest/connecting_data/Datafeed-Issues#getbars-is-called-multiple-times).

>Does any Android library exist that for advanced trading view library ?

[Here](https://github.com/tradingview/charting-library-examples/tree/master/android) we have a dedicated example for Android using webviews.
Basically the library is in Javascript to ease its usability in most framework/environment.
